src/pymor/models/transfer_function.py

```diff
@@ -271,30 +288,38 @@ def bode_plot(self, w, mu=None, ax=None, Hz=False, dB=False, deg=True, adaptive_
         freq = freq / self.sampling_time if self.sampling_time > 0 else freq
         if deg:
             phase *= 180 / np.pi
-
         artists = np.empty_like(ax)
         freq_label = f'Frequency ({"Hz" if Hz else "rad/s"})'
         mag_label = f'Magnitude{" (dB)" if dB else ""}'
         phase_label = f'Phase ({"deg" if deg else "rad"})'
-        for i in range(self.dim_output):
-            for j in range(self.dim_input):
+        for i in range(num_output):
+            for j in range(num_input):
                 if dB:
-                    artists[2 * i, j] = ax[2 * i, j].semilogx(freq, 20 * np.log10(mag[:, i, j]),
-                                                              **mpl_kwargs)
+                    artists[2 * i, j] = ax[2 * i, j].semilogx(freq, 20 * np.log10(mag[:, output_indices[i],
+                                                              input_indices[j]]), **mpl_kwargs)
                 else:
-                    artists[2 * i, j] = ax[2 * i, j].loglog(freq, mag[:, i, j],
-                                                            **mpl_kwargs)
-                artists[2 * i + 1, j] = ax[2 * i + 1, j].semilogx(freq, phase[:, i, j],
-                                                                  **mpl_kwargs)
-        for i in range(self.dim_output):
+                    artists[2 * i, j] = ax[2 * i, j].loglog(freq, mag[:, output_indices[i],
+                                                            input_indices[j]], **mpl_kwargs)
+                artists[2 * i + 1, j] = ax[2 * i + 1, j].semilogx(freq, phase[:, output_indices[i],
+                                                                  input_indices[j]], **mpl_kwargs)
+        for i in range(num_output):
             ax[2 * i, 0].set_ylabel(mag_label)
             ax[2 * i + 1, 0].set_ylabel(phase_label)
-        for j in range(self.dim_input):
+        for j in range(num_input):
             ax[-1, j].set_xlabel(freq_label)
         fig.suptitle('Bode plot')
-
         return artists
 
+    @staticmethod
+    def _cal_plot_dimension(indices, dim):
+        if indices is None:
+            return dim
+        else:
+            assert all(isinstance(item, int) for item in indices)
+            assert all(-dim <= item < dim for item in indices), \
+                f'Indices should be any integer value between {-dim} and {dim-1}'
```

Comment:
Negative indices are allowed because `numpy` treats them by counting backwards, right? Maybe it should also be mentioned in the docstring that this behavior can be used.

Reply:
Almost any iterables in python can be access by giving negative indices for example python `list` and `tuple` can accept negative indices so it not only `numpy`. But I will mention this in the docstring

Reply:
Yes, I know, I was just wondering whether this is the reason for having negative indices. But seems reasonable to me and might be useful sometimes.

Reply:
@HenKlei, I think it will be a good thing to add subtitle to the plots as it was confusing for me which input and which output is being displayed. But the inputs and the outputs does not have a unique name so what do you suggest as a subtitle ?